src/debugger/appWorker.ts

```diff
@@ -203,6 +206,9 @@ export class MultipleLifetimesAppWorker {
             if (object.method === "prepareJSRuntime") {
                 // The MultipleLifetimesAppWorker will handle prepareJSRuntime aka create new lifetime
                 this.gotPrepareJSRuntime(object);
+            } else if (object.method === "$disconnected") {
+                // We need to shutdown the current app worker, and create a new lifetime
+                this.singleLifetimeWorker = null;
```

Comment:
The difficulty with using cluster like that is the debugging; the different members of the cluster are different processes and so we won't be debugging the thing that runs the code any more. We could manually do something similar though where we download the bundle.js and bundle.map.js and then explicitly fork a node process and attach to it for debugging.